src/tests/ftest/recovery/check_policy.py

```diff
@@ -0,0 +1,165 @@
+        policies = self.check_all_default(dmg_command=dmg_command)
+
+        # 4. Set invalid policy to some policy. Verify error message.
+        self.log_step("Set invalid policy to some policy. Verify error message.")
+        class_name = "POOL_NONEXIST_ON_MS"
+        invalid_policy = "CIA_X"
+        invalid_class_policy = class_name + ":" + invalid_policy
+        errors = []
+        try:
+            dmg_command.check_set_policy(policies=invalid_class_policy)
+        except CommandFailure as command_failure:
+            exp_msg = f'invalid inconsistency action "{invalid_policy}"'
+            if exp_msg not in str(command_failure):
+                msg = f"Setting invalid policy didn't return expected message! {exp_msg}"
+                errors.append(msg)
+            else:
+                self.log.info("Setting invalid policy failed as expected.")
+
+        # 5. Set CIA_INTERACT to some policies.
+        self.log_step("Set CIA_INTERACT to some policies.")
+        interact_count = 4
+        for i in range(interact_count):
+            class_name = policies[i].split(":")[0]
+            class_policy = class_name + ":" + "CIA_INTERACT"
+            dmg_command.check_set_policy(policies=class_policy)
+
+        # 6. Set CIA_INTERACT to one of the policies from the previous step.
+        self.log_step("Set CIA_INTERACT to one of the policies from the previous step.")
+        class_name = policies[0].split(":")[0]
+        class_policy = class_name + ":" + "CIA_INTERACT"
+        dmg_command.check_set_policy(policies=class_policy)
+
+        # 7. Verify that INTERACT is set to the policies from the previous step. Also
+        # verify that the rest of the policies are still DEFAULT.
+        msg = ("Verify that INTERACT is set to the policies from the previous step. "
+               "Also verify that the rest of the policies are still DEFAULT.")
+        self.log_step(msg)
+        self.check_interact(dmg_command=dmg_command, interact_count=interact_count)
+
+        # 8. Disable and enable. Verify that the same policies are still there.
+        self.log_step(
+            "Disable and enable. Verify that the same policies are still there.")
+        dmg_command.check_disable()
+        dmg_command.check_enable()
+        self.check_interact(dmg_command=dmg_command, interact_count=interact_count)
+
+        # 9. Reset with --reset-defaults. Verify that all policies are back to DEFAULT.
+        msg = "Reset with --reset-defaults. Verify that all policies are back to DEFAULT."
+        self.log_step(msg)
+        dmg_command.check_set_policy(reset_defaults=True)
+        self.check_all_default(dmg_command=dmg_command)
+
+        # 10. Start the checker. It should automatically repair the fault.
+        self.log_step("Start the checker. It should automatically repair the fault.")
+        dmg_command.check_start()
+        # Verify that the orphan pool is fixed by querying the checker.
+        self.wait_for_check_complete()
```

Comment:
In https://github.com/daos-stack/daos/pull/16656 this will no longer be a `RecoveryTestBase` method. Instead, it is imported from `src/tests/ftest/util/recovery_utils.py` and called like so: 
```suggestion
        wait_for_check_complete(dmg_command)
```